tracing-subscriber/src/registry/mod.rs

```diff
@@ -250,18 +264,42 @@ where
         })
     }
 
-    /// Returns an iterator over all parents of this span.
+    /// Returns an iterator over all parents of this span, starting with the
+    /// immediate parent.
     ///
     /// The iterator will first return the span's immediate parent, followed by
     /// that span's parent, followed by _that_ span's parent, and so on, until a
     /// it reaches a root span.
-    pub fn parents(&self) -> Parents<'_, R> {
+    pub fn parents(&self) -> Parents<'a, R> {
         Parents {
             registry: self.registry,
             next: self.parent().map(|parent| parent.id()),
         }
     }
 
+    /// Returns an iterator over all parents of this span, starting with the
+    /// root of the trace tree.
+    ///
+    /// The iterator will return the root of the trace tree, followed by the
+    /// next span, and then the next, until this span's immediate parent is
+    /// returned.
+    ///
+    /// **Note**: if the "smallvec" feature flag is not enabled, this may
+    /// allocate.
+    pub fn from_root(&self) -> FromRoot<'a, R> {
```

Comment:
I _love_ this. If I recall correctly, @pkinsky expressed interest in this!

Reply:
I did, thanks!